### PR TITLE
Add more JPEG XL tests

### PR DESCRIPTION
**Pull request by mo271**

### CODE REVIEW

File: jpegxl/resources/generate_resources.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  convert_and_compress 3x3.png "3x3" "$color_space"
+  convert_and_compress 3x3a.png "3x3a" "$color_space"
+done
+
+convert 3x3.png -quality 100 3x3.jpg
+# lossless recompression
+cjxl 3x3.jpg 3x3_jpeg_recompression.jxl
+# checking that it was actually byte exact
+djxl 3x3_jpeg_recompression.jxl 3x3_recoverd.jpg
+diff 3x3.jpg 3x3_recoverd.jpg
```
Comment (foolip) [inline jpegxl/resources/generate_resources.sh:67]: Fail if there was a diff?

File: jpegxl/resources/generate_resources.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# generate golden png
```
Comment (foolip) [inline jpegxl/resources/generate_resources.sh:68]: Using more WPTy vocabulary:

```suggestion
# generate reference png
```

File: jpegxl/resources/generate_resources.sh (hunk 1 above)
Comment (foolip) [reply jpegxl/resources/generate_resources.sh:67]: Like this:

```suggestion
diff -q 3x3.jpg 3x3_recovered.jpg || exit 1
```

File: jpegxl/resources/generate_resources.sh [hunk 3]
```diff
@@ -59,13 +59,13 @@ for color_space in "${color_spaces[@]}"; do
   convert_and_compress 3x3a.png "3x3a" "$color_space"
 done
 
-convert 3x3.png -quality 100 3x3.jpg
+convert 3x3.png -quality 70 3x3.jpg
 # lossless recompression
 cjxl 3x3.jpg 3x3_jpeg_recompression.jxl
 # checking that it was actually byte exact
 djxl 3x3_jpeg_recompression.jxl 3x3_recoverd.jpg
```
Comment (foolip) [inline jpegxl/resources/generate_resources.sh:66]:
```suggestion
djxl 3x3_jpeg_recompression.jxl 3x3_recovered.jpg
```

File: jpegxl/resources/generate_resources.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+# lossless recompression
+cjxl 3x3.jpg 3x3_jpeg_recompression.jxl
+# checking that it was actually byte exact
+djxl 3x3_jpeg_recompression.jxl 3x3_recoverd.jpg
+diff 3x3.jpg 3x3_recoverd.jpg
+# generate reference png
+djxl 3x3_jpeg_recompression.jxl 3x3_jpeg_recompression.png
+
+# Cleanup temporary file
+rm -f temp.jxl 3x3.png 3x3a.png 3x3.jpg 3x3_recoverd.jpg
```
Comment (foolip) [inline jpegxl/resources/generate_resources.sh:72]:
```suggestion
rm -f temp.jxl 3x3.png 3x3a.png 3x3.jpg 3x3_recovered.jpg
```

File: jpegxl/resources/generate_resources.sh (hunk 1 above)
Comment (foolip) [reply jpegxl/resources/generate_resources.sh:67]: Your way is better :)

Comment (mo271) [reply jpegxl/resources/generate_resources.sh:67]: did it differently...

**Review by foolip — APPROVED** [review]

**Pull request merged**